src/keypair.rs

```diff
@@ -431,6 +434,83 @@ impl Verifier<ed25519::Signature> for Keypair {
     }
 }
 
+impl TryFrom<&[u8]> for Keypair {
+    type Error = SignatureError;
+
+    fn try_from(bytes: &[u8]) -> Result<Keypair, SignatureError> {
+        Keypair::from_bytes(bytes)
```

Comment:
Is there a reason to keep the `Keypair::from_bytes()` methods anymore? Documentation on methods of trait impls works how we'd want it to. Small other change is `SecretKey::from_bytes -> try_from` for consistency.

Reply:
Yes, per #230 it can be changed to accept a fixed-sized array or reference thereof, which can catch errors related to the input size at compile time rather than runtime. The `TryFrom<&[u8]>` impls can still accept a slice for convenience (and to keep the type signatures there simple).

As it were, a local WIP PR which I haven't pushed up yet which renames `Keypair` -> `SigningKey` and `PublicKey` -> `VerifyingKey` does just that.

Reply:
Gotcha makes sense